tests/Functional/SelfUpdateTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Manalize project.
+ *
+ * (c) Manala <contact@manala.io>
+ *
+ * For the full copyright and license information, please refer to the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Manala\Manalize\Tests\Functional;
+
+use Manala\Manalize\Command\Setup;
+use Manala\Manalize\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\Process;
+
+class SelfUpdateTest extends \PHPUnit_Framework_TestCase
+{
+    private static $cwd;
+
+    public static function setUpBeforeClass()
+    {
+        $cwd = sys_get_temp_dir().'/Manala';
+        $fs = new Filesystem();
+
+        if ($fs->exists($cwd)) {
+            $fs->remove($cwd);
+        }
+
+        $fs->mkdir($cwd);
+
+        (new Process('make build', getcwd()))->run();
+        chmod(getcwd().'/manalize.phar', 0777);
+        rename(getcwd().'/manalize.phar', $cwd.'/manalize.phar');
+
+        self::$cwd = $cwd;
+    }
+
+    public function testExecute()
+    {
+        $latestBuild = json_decode(file_get_contents(
+            sprintf('https://api.github.com/repos/%s/releases/latest', Application::REPOSITORY_NAME),
+            null,
+            stream_context_create(['http' => ['header' => 'User-Agent: '.Application::REPOSITORY_NAME]])
+        ), true);
+        $latestTag = $latestBuild['tag_name'];
+
+        if ('v'.Application::VERSION === $latestTag) {
+            $this->markTestSkipped('The current version is the latest one, unable to update it');
+        }
+
+        $process = new Process('cd '.self::$cwd.' && ./manalize.phar self-update');
```

Comment:
```diff
- $process = new Process('cd '.self::$cwd.' && ./manalize.phar self-update');
+ $process = new Process(self::$cwd.'/manalize.phar self-update');
```

This ensure the selfupdate process does not depend in any way on the current working dir.